src/ott/problems/linear/potentials.py

```diff
@@ -157,6 +161,144 @@ def tree_unflatten(
+        if ``True``, otherwise use the inverse map
+      ax: axis to add the plot to
+      alpha: transparency of the scatter plot points
+      scatter_kwargs: additional kwargs passed into :meth:`~matplotlib.axes.Axes.scatter`
+      legend_kwargs: additional kwargs passed into :meth:`~matplotlib.axes.Axes.legend`
+
+    Returns: matplotlib figure and axis with the plots
+    """
+    if scatter_kwargs is None:
+      scatter_kwargs = {}
+    if legend_kwargs is None:
+      legend_kwargs = {}
+
+    if ax is None:
+      fig = plt.figure(facecolor="white")
+      ax = fig.add_subplot(111)
+    else:
+      fig = ax.get_figure()
+
+    # plot the source and target samples
+    if forward:
+      label_transport = r"$\nabla f(source)$"
+      source_color, target_color = "#1A254B", "#A7BED3"
+    else:
+      label_transport = r"$\nabla g(target)$"
+      source_color, target_color = "#A7BED3", "#1A254B"
+
+    ax.scatter(
+        source[:, 0],
+        source[:, 1],
+        color=source_color,
+        alpha=alpha,
+        label='source',
+        **scatter_kwargs,
+    )
+    ax.scatter(
+        target[:, 0],
+        target[:, 1],
+        color=target_color,
+        alpha=alpha,
+        label='target',
+        **scatter_kwargs,
+    )
+
+    # plot the transported samples
+    base_samples = source if forward else target
+    transported_samples = self.transport(base_samples, forward=forward)
+    ax.scatter(
+        transported_samples[:, 0],
+        transported_samples[:, 1],
+        color="#F2545B",
+        alpha=alpha,
+        label=label_transport,
+        **scatter_kwargs,
+    )
+
+    for i in range(base_samples.shape[0]):
+      ax.arrow(
+          base_samples[i, 0],
+          base_samples[i, 1],
+          transported_samples[i, 0] - base_samples[i, 0],
+          transported_samples[i, 1] - base_samples[i, 1],
+          color=[0.5, 0.5, 1],
+          alpha=0.3
+      )
+
+    ax.legend(**legend_kwargs)
+    return fig, ax
+
+  def plot_potential(
+      self,
+      forward: bool = True,
+      alpha: float = 0.05,
+      ax: Optional[matplotlib.axes.Axes] = None,
+      x_bounds: Tuple[float, float] = (-6, 6),
+      y_bounds: Tuple[float, float] = (-6, 6),
+      num_grid: int = 50,
+      contourf_kwargs: Optional[Dict[str, Any]] = None,
+  ) -> Tuple[matplotlib.figure.Figure, matplotlib.axes.Axes]:
+    """Plot the potential.
+
+    Args:
+      forward: use the forward map from the potentials
+        if ``True``, otherwise use the inverse map
+      alpha: quantile to filter the potentials with
+      ax: axis to add the plot to
+      x_bounds: x-axis bounds of the plot (xmin, xmax)
+      y_bounds: y-axis bounds of the plot (ymin, ymax)
+      num_grid: number of points to discretize the domain into a grid
+        along each dimension
+      contourf_kwargs: additional kwargs passed into :meth:`~matplotlib.axes.Axes.contourf`
+
+    Returns: matplotlib figure and axis with the plots.
```

Comment:
Can you put `matplotlib ...` on new line + indent? Think this is why it renders weird in the docs.